cpu/stm32/kconfigs/Kconfig.clk

```diff
@@ -208,15 +260,28 @@ config CLOCK_MSI
     default 131072 if CLOCK_MSI_130KHZ
     default 262144 if CLOCK_MSI_260KHZ
     default 524288 if CLOCK_MSI_520KHZ
-    default 1048000 if CLOCK_MSI_1MHZ
-    default 2097000 if CLOCK_MSI_2MHZ
-    default 4194000 if CLOCK_MSI_4MHZ
-
-endif  # CPU_FAM_L0 || CPU_FAM_L1
+    default 100000 if CLOCK_MSI_100KHZ
+    default 200000 if CLOCK_MSI_200KHZ
+    default 400000 if CLOCK_MSI_400KHZ
+    default 800000 if CLOCK_MSI_800KHZ
+    default 1000000 if CLOCK_MSI_1MHZ && (CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB)
+    default 1048000 if CLOCK_MSI_1MHZ && (CPU_FAM_L0 || CPU_FAM_L1)
+    default 2000000 if CLOCK_MSI_2MHZ && (CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB)
+    default 2097000 if CLOCK_MSI_2MHZ && (CPU_FAM_L0 || CPU_FAM_L1)
+    default 4000000 if CLOCK_MSI_4MHZ && (CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB)
+    default 4194000 if CLOCK_MSI_4MHZ && (CPU_FAM_L0 || CPU_FAM_L1)
```

Comment:
I made a suggestion on the prompts for these options, but alternatively we could split them in different choice options altogether.

Reply:
I went for your suggestion that I find simpler.